cypress/integration/e2e/consent.spec.js

```diff
@@ -0,0 +1,116 @@
+import { getPolyfill } from '../../lib/polyfill';
+import { fetchPolyfill } from '../../lib/config';
+
+const firstPage =
+    'https://www.theguardian.com/environment/2020/oct/13/maverick-rewilders-endangered-species-extinction-conservation-uk-wildlife';
+
+const secondPage =
+    'https://www.theguardian.com/environment/2020/nov/19/blue-whale-sightings-off-south-georgia-raise-hopes-of-recovery';
+
+describe('Consent tests', function () {
+    before(getPolyfill);
+
+    const cmpIframe = () => {
+        return cy
+            .get('iframe[id*="sp_message_iframe"]')
+            .its('0.contentDocument.body')
+            .should('not.be.empty')
+            .then(cy.wrap);
+    };
+
+    const privacySettingsIframe = () => {
+        return cy
+            .get('[src*="https://cdn.privacy-mgmt.com/privacy-manager"]')
+            .its('0.contentDocument.body')
+            .should('not.be.empty')
+            .then(cy.wrap);
+    };
+
+    beforeEach(function () {
+        // Reset CMP cookies before each test
+        cy.clearCookie('consentUUID');
+        cy.clearCookie('euconsent-v2');
+        // Setup a listener for calls to Google Analytics
+        cy.server();
+        cy.route({
+            method: 'POST',
+            url: 'https://www.google-analytics.com/**',
+        }).as('theCallToGoogle');
+    });
+
+    it('should make calls Google Analytics after the reader consents', function () {
+        cy.visit(`Article?url=${firstPage}`, fetchPolyfill);
+        // Open the Privacy setting dialogue
+        cmpIframe().contains("It's your choice");
+        cmpIframe().find("[title='Manage my cookies']").click();
+        // Accept tracking cookies
+        privacySettingsIframe().contains('Privacy settings');
+        privacySettingsIframe().find("[title='Accept all']").click();
+        // Make a second page load now that we have the CMP cookies set to accept tracking
+        cy.visit(`Article?url=${secondPage}`, fetchPolyfill);
+        // Wait for a call to Google Analytics to be made - we expect this to happen
+        cy.wait('@theCallToGoogle', { timeout: 3000 });
+    });
+
+    /**
```

Comment:
@oliverlloyd went down a rabbit hole trying to write Cypress tests we were happy with for the case we care most about (not tracking users who have not yet given their consent, or have removed their consent). We'd like to include the tests as comments in case someone wants to build on these at a later date. They work, but as outlined here they are slow, and possibly fail to verify that events do not fire. Any objections?